packages/launchpad/cypress/e2e/migration.cy.ts

```diff
@@ -1734,45 +1734,103 @@ describe('Migrate custom config files', () => {
 })
 
 describe('v13 migration welcome page with video', () => {
-  function stubVideoHtml (): void {
+  it('Welcome page should appear if video is not present (failure)', () => {
     cy.withCtx((ctx, o) => {
+      const originalGetVideoEmbedHtml = ctx.migration.getVideoEmbedHtml
+
       o.sinon.stub(ctx.migration, 'getVideoEmbedHtml').callsFake(async () => {
-        return '<span>Stubbed Video Content</span>'
+        const mockMigrationSourceGetVideoEmbedHtmlCTX = {
+          ctx: {
+            coreData: {
+              migration: {
+                videoEmbedHtml: undefined,
+              },
+            },
+            versions: {
+              versionData: () => {
+                return {
+                  current: {
+                    version: '13.0.0',
+                  },
+                }
+              },
+            },
+            util: {
+              fetch: () => {
+                throw new Error('kaboom')
+              },
+            },
+          },
+        }
+
+        return originalGetVideoEmbedHtml.apply(mockMigrationSourceGetVideoEmbedHtmlCTX)
       })
     })
-  }
 
-  function unstubVideoHtml (): void {
-    cy.withCtx((ctx, o) => {
-      const restoreFn = (ctx.migration.getVideoEmbedHtml as SinonStub).restore
-
-      restoreFn?.()
-    })
-  }
+    cy.scaffoldProject('migration-v12-to-v13')
+    cy.openProject('migration-v12-to-v13')
 
-  beforeEach(() => {
-    stubVideoHtml()
+    cy.visitLaunchpad()
+    cy.contains(cy.i18n.majorVersionWelcome.title).should('be.visible')
+    cy.get('[data-cy="video-container"]').should('not.exist')
   })
 
-  it('Welcome page should appear if video is not present', () => {
-    unstubVideoHtml()
-
-    cy.scaffoldProject('migration-v12-to-v13')
-    cy.openProject('migration-v12-to-v13')
+  it('Welcome page should appear if video is not present (timeout)', () => {
     cy.withCtx((ctx, o) => {
+      const originalGetVideoEmbedHtml = ctx.migration.getVideoEmbedHtml
+
       o.sinon.stub(ctx.migration, 'getVideoEmbedHtml').callsFake(async () => {
-        return null
+        const mockMigrationSourceGetVideoEmbedHtmlCTX = {
```

Comment:
need to duplicate the code here since withCtx does not have hoisting capabilities (similar to cy.origin())